cases/security_test.py

```diff
@@ -339,6 +422,38 @@ def on_done(_: Any) -> None:
                 dut_pairing.cancel()
                 ref_pairing.cancel()
 
+        if not need_ctkd:
+            return
+
+        ctkd_shall_pass = variant == 'accept_ctkd'
+
+        if variant == 'accept_ctkd':
+
+            async def ctkd_over_bredr() -> None:
+                if ref_role == 'against_central':
+                    if ref_dut_bumble is not None:
+                        await ref_dut_bumble.pair()
+                else:
+                    if dut_ref_bumble is not None:
+                        await dut_ref_bumble.pair()
+                assert ctkd_task is not None
+                await ctkd_task
+
+            await ctkd_over_bredr()
+        else:
+            fail("Unsupported variant " + variant)
+
+        if ctkd_shall_pass:
+            if ref_dut_bumble is not None:
+                ref_dut_keys = await self.ref.device.keystore.get(str(ref_dut_bumble.peer_address))  # type: ignore
+                assert_is_not_none(ref_dut_keys.ltk)  # type: ignore
+                assert_is_not_none(ref_dut_keys.irk)  # type: ignore
+                assert_is_not_none(ref_dut_keys.csrk)  # type: ignore
+
+            # Address resolving doesn't work for now
+            # TODO: Fix Pandora/Bumble/RootCanal address resolving
+            # await le_connect_with_rpa_and_encrypt(self.dut, self.ref)
```

Comment:
What do you think is going on ?

Reply:
1. Without https://github.com/google/bumble/pull/231, Bumble always distributes IRK with random identity address and I am not pretty sure what's its impact on Android
2. It seems Bumble doesn't really use resolable random address. It simply set address once when power on https://github.com/google/bumble/blob/main/bumble/device.py#L1252. Address resolving is also disabled in https://github.com/google/bumble/blob/main/bumble/device.py#L1277C23-L1277C67

Reply:
Fix:
https://android-review.git.corp.google.com/c/platform/packages/modules/Bluetooth/+/2688968
https://github.com/google/bumble/pull/234